src/types/helperTypes.d.ts

```diff
@@ -161,3 +161,95 @@ export type DataSourceForPlaceable<P extends PlaceableObject> = P extends Placea
     ? D["_source"]
     : never
   : never;
+
+type EndValue = string | bigint | number | boolean | null | undefined;
+export type InputValue = object | Array<unknown>;
```

Comment:
`object | Array<unknown>` is redundant as `object` includes arrays. I'd do `Record<string, unknown> | Array<unknown>` for essentially clarity. Also I'd probably not make this exported.